middleware/2fa.js

```diff
@@ -0,0 +1,287 @@
+    await twoFactorMethod.update({ securityCode, requestId });
+    const dataOutput = data.request ? prettyRequestInfo(data) : `Verifying ${method.detail} as 2FA method`;
+    const text = 
+`
+NEAR Wallet security code: ${securityCode}\n\n
+Important: By entering this code, you are authorizing the following transaction:\n\n
+${dataOutput}
+`;
+    const html =
+`
+<body style="margin: 0; padding: 0;">
+    <table align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+        <tr>
+            <td align="center">
+                <img src="https://near.org/wp-content/themes/near-19/assets/downloads/near_logo.png" width="300" height="117"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <p>NEAR Wallet security code: ${securityCode}</p>
+                <p><strong>Important:</strong> By entering this code, you are authorizing the following transaction:\n\n</p>
+                <pre>
+                    ${dataOutput}
+                </pre>
+            </td>
+        </tr>
+    </table>
+</body>
+`;
+    if (method.kind === '2fa-phone') {
+        await sendSms({
+            text,
+            to: method.detail
+        });
+    } else if (method.kind === '2fa-email') {
+        await sendMail({
+            to: method.detail,
+            subject: `NEAR Wallet security code: ${securityCode}`,
+            text,
+            html
+        });
+    }
+    return securityCode;
+};
+
+/********************************
+Checking code_hash (is multisig deployed)
+********************************/
+const isContractDeployed = async(accountId) => {
+    const keyStore = {
+        async getKey() {
+            return nearAPI.KeyPair.fromString('bs');
+        },
+    };
+    // check account code_hash
+    const near = await nearAPI.connect({
+        deps: { keyStore },
+        masterAccount: creatorKeyJson && creatorKeyJson.account_id,
+        nodeUrl: process.env.NODE_URL
+    });
+    const nearAccount = new nearAPI.Account(near.connection, accountId);
+    const state = await nearAccount.state();
+    console.log(state);
+    return state.code_hash !== '11111111111111111111111111111111';
+};
+
+/********************************
+@warn protect these routes using checkAccountOwnership middleware from app.js
+@warn Requires refactor
+********************************/
+// http post http://localhost:3000/2fa/getAccessKey accountId=mattlock
+// http post https://helper.testnet.near.org/2fa/getAccessKey accountId=mattlock
+// http post https://near-contract-helper-2fa.onrender.com/2fa/getAccessKey accountId=mattlock
+// Call this to get the public key of the access key that contract-helper will be using to confirm multisig requests
+const getAccessKey = async (ctx) => {
+    const { accountId } = ctx.request.body;
+    const { publicKey } = await getDetermKey(accountId);
+    ctx.body = {
+        success: true,
```

Comment:
why is `success: true` needed? HTTP already has success / non-success statuses: https://httpstatuses.com

Reply:
Habit. I'll check with @Patrick1904 if this is baked into frontend and cleanup if easy